### PR TITLE
Happy Thoughts API, Hannah Jesinkey

### PR DESCRIPTION
**Pull request by jesinkey**
Deployed app: https://happy-thoughts-hannahj.herokuapp.com/

### CODE REVIEW

**Review by jessika-hage — APPROVED** [review]
I don't have anything to comment on other than that the code looks very nice and it is easy to read and understand! Great job! :D

File: README.md [hunk 1]
```diff
@@ -4,10 +4,13 @@ Replace this readme with your own information about your project.
 
 Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
 
+Create the backend for our Happy Thoughts Project. We were to create different endpoints using post and get to show all of our messages as well as add new ones and increase likes on those we had.
+
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+A lot this week was just following the lectures and code-along. Each step was very straight forward and easy to follow I thought.
+If I had more time I would have created a delete-button for my frontend to match the delete-function in my backend.
```
Comment (KarolCislowski) [inline README.md:13]: some documentation on how to use your API would be nice 🤓

File: server.js [hunk 2]
```diff
@@ -1,30 +1,112 @@
...
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.set('useCreateIndex', true);
+mongoose.Promise = Promise;
+
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, 'A Happy Thought is required'],
+    minlength: 5,
+    maxlength: 140,
```
Comment (KarolCislowski) [inline server.js:16]: Nicely configured field 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,30 +1,112 @@
...
-})
+  res.send(listEndpoints(app));
+});
+
+// Endpoint to display all Happy Thoughts
+app.get('/thoughts', async (req, res) => {
+  const allThoughts = await Thought.find()
+    .sort({ createdAt: 'desc' })
+    .limit(20)
+    .exec();
```
Comment (KarolCislowski) [inline server.js:46]: 💯

File: server.js [hunk 4]
```diff
@@ -1,30 +1,112 @@
...
+        _id: id,
+      },
+      {
+        $inc: {
+          hearts: 1,
+        },
+      },
+      {
+        new: true,
+      }
```
Comment (KarolCislowski) [inline server.js:81]: 💯

File: server.js [hunk 5]
```diff
@@ -1,30 +1,112 @@
...
+    const deletedThought = await Thought.findOneAndDelete({ _id: id });
+    if (deletedThought) {
+      res.json(deletedThought);
+    } else {
+      res.status(404).json({ message: 'Not found' });
+    }
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error });
+  }
+});
```
Comment (KarolCislowski) [inline server.js:107]: remember to restrict endpoints that can manipulate with data stored in your DB 🤓

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Your code looks nice but you need to provide a link to the frontend connected to this API for testing 😸

**Comment by jesinkey** [comment]
Sorry! Here is the frontend link: https://happy-thoughts-hj.netlify.app/

**Review by KarolCislowski — APPROVED** [review]
Great App

**Pull request closed without merging**